### ISSUE TITLE
Better control in fzf searching

### ISSUE THREAD

**Issue by khaosdoctor** [aome510/spotify-player#641]
**Is your feature already implemented in the latest `master`?** No

**Is your feature request related to a problem? Please describe.**
Being able to use fzf with `/` to find tracks in the liked songs (`gy`), for example, is an awesome addition. The problem is that once searched, I can't do anything other than play the track. If I want to add the searched track to the queue, this is not possible because every command will be typed in the search field. I've tried `C-space`, change focus, but nothing works. The only key that works is `enter`, if I press `C-space` I'm thrown back at the top of my liked songs list.

**Describe the solution you'd like**
It would be nice if the fzf search for every page worked like the search page, e.g: press `/` a new field pops below the current playing song progress bar, you can type in and the list is being filtered as you type. But, as you press `FocusNextPane` you drop to the results list where you can apply the same commands as you would be able to in the liked songs list or the search results list

**Describe alternatives you've considered**
What I'm doing today is that I use the search page to search for the song I want, even though I know the song is already saved in my library and then I use `Z` to add to the queue.

**Additional context**
Version: 0.20.4
MacOS Sequoia 15.1.1
Installed through Cargo with all the features

**Comment by aome510** [comment]
> The only key that works is enter, if I press C-space I'm thrown back at the top of my liked songs list.

Yes, but it should show a popup for you to initiate an action on the selected track. 

> If I want to add the searched track to the queue, this is not possible because every command will be typed in the search field

No, not every command. Commands with key bindings `control + {char}` or `alt + {char}` still work in the search. FYI, `AddSelectedItemToQueue` does have a `C-z` (control+z) binding.

Search is intended to be **interactive**. The solution you'd like to see defeats that purpose

**Comment by khaosdoctor** [comment]
I see thanks for the clarification, in any case there’s the C-space bug, if I press it, the interactive search goes away and I’m thrown back at the top of the list.

Regarding the fzf search, would it be possible to add the binding to move to the next pane and unfocus the search box? This would solve all the issues

**Comment by wiraki** [comment]
Just started using spotify-player, it's awesome! Thank you for making it.

It's amazing to have fzf support, but in my first hour of using it I did struggle with the same problem described in this issue.

> Yes, but it should show a popup for you to initiate an action on the selected track.

That's for the top result of the fzf-search, which will be considered as "selected". Which means that I have to carefully adjust my query to make sure the track I am looking for and I want to interact with be the top result. I would argue this beats the purpose of fuzzy searching.

In my mind, after popping up the search, typing a few characters (might even make some typos) and filtering down the items, it would then make sense to be able to "browse" through the search results, before taking any action.

@khaosdoctor said:
>Regarding the fzf search, would it be possible to add the binding to move to the next pane and unfocus the search box? This would solve all the issues

Indeed, this would be a solution and initially I was expecting this to be possible from reading the documentation:

> To move the focus from the search input to the other windows such as track results, album results, etc, use FocusNextWindow or FocusPreviousWindow.

But that is not true when fuzzy search is enabled apparently? Is that right? If so, a clarification would be good in the documentation. (I can make the contribution when I completely understand how the fuzzy search works).

As a workaround, I found that after fuzzy searching I can use the arrow keys to navigate the search results.

**Comment by juliamertz** [comment]
I ran into the same problem, so I've added an option for modal input in https://github.com/juliamertz/spotify-player/commit/afe2bc61cca6db62bddb3a57b3f6a6be20ed8f20

If the option is enabled, you're put into insert mode when searching. Then if you press escape (or enter to submit the query), you're put into normal mode so you can browse the results pane and use all the keybinds.

I wrote this a while ago, but if there's interest, I can finish it up and open a PR.
